tests/test_internals/test_si.py

```diff
@@ -576,5 +579,74 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
+class TestSpikeInterfaceRecorders(unittest.TestCase):
+    def setUp(self):
+
+        self.test_dir = tempfile.mkdtemp()
+        self.path1 = self.test_dir + "/test_electrodes1.nwb"
+
+        self.nwbfile1 = NWBFile("session_description1", "file_id1", datetime.now())
+
+        self.RX1 = generate_recording(num_channels=4, durations=[3])
+        self.metadata_list = [dict(Ecephys={i: dict(name=i, description="desc")}) for i in ["es1", "es2"]]
+
+    def tearDown(self):
+        del self.RX1
+        shutil.rmtree(self.test_dir)
```

Comment:
One thing to learn about `TestCase` is the difference between available setup methods. `setUp` and `tearDown` run each time a test function in the class is run. Using

```python
    @classmethod
    def setUpClass(cls):
        ...

    @classmethod
    def tearDownClass(cls):
        ...
```

instead only run once total for the testing class. Try to think about what setup operations can be called fewer times using this distinction vs. what 'new'/'fresh' objects you need to have going into each test for custom modification leading to assertion.

Reply:
This is good, I was not aware of this.